src/Compilers/CSharp/Test/Semantic/Semantics/InterceptorsTests.cs

```diff
@@ -4892,16 +4891,298 @@ public static void Main()
+                internal static void Interceptor(this C c) => Console.Write(1);
+            }
+            """;
+
+        var comp = CreateCompilation(new[] { (source, PlatformInformation.IsWindows ? @"C:\src\Program.cs" : "/src/Program.cs"), (source2, PlatformInformation.IsWindows ? @"C:\obj\Generated.cs" : "/obj/Generated.cs"), s_attributesSource }, parseOptions: RegularWithInterceptors);
+        comp.VerifyEmitDiagnostics();
+    }
+
+    [Fact]
+    public void RelativePaths_06()
+    {
+        var source = """
+            class C
+            {
+                public static void Main()
+                {
+                    C c = new C();
+                    c.M();
+                }
+
+                public void M() => throw null!;
+            }
+            """;
+
+        var source2 = """
+            using System;
+            using System.Runtime.CompilerServices;
+
+            static class Interceptors
+            {
+                [InterceptsLocation("../src/Program.cs/.", 6, 11)]
+                internal static void Interceptor(this C c) => Console.Write(1);
+            }
+            """;
+
+        var comp = CreateCompilation(new[] { (source, PlatformInformation.IsWindows ? @"C:\src\Program.cs" : "/src/Program.cs"), (source2, PlatformInformation.IsWindows ? @"C:\obj\Generated.cs" : "/obj/Generated.cs"), s_attributesSource }, parseOptions: RegularWithInterceptors);
+        comp.VerifyEmitDiagnostics();
+    }
+
+    [Fact]
+    public void RelativePaths_07()
+    {
+        var source = """
+            C c = new C();
+            c.M();
+
+            class C
+            {
+                public void M() => throw null!;
+            }
+            """;
+
+        var source2 = """
+            using System.Runtime.CompilerServices;
+            using System;
+
+            static class Interceptors
+            {
+                [InterceptsLocation("../src/Program.cs", 2, 3)]
+                public static void Interceptor(this C c) => Console.Write(1);
+            }
+            """;
+        var pathPrefix = PlatformInformation.IsWindows ? """C:\My\Machine\Specific\Path\""" : "/My/Machine/Specific/Path/";
+        var path = pathPrefix + "src/Program.cs";
+        var path2 = pathPrefix + "obj/Generated.cs";
+        var pathMap = ImmutableArray.Create(new KeyValuePair<string, string>(pathPrefix, "/_/"));
+
+        var verifier = CompileAndVerify(
+            new[] { (source, path), (source2, path2), s_attributesSource },
+            parseOptions: RegularWithInterceptors,
+            options: TestOptions.DebugExe.WithSourceReferenceResolver(
+                new SourceFileResolver(ImmutableArray<string>.Empty, null, pathMap)),
+            expectedOutput: "1");
+        verifier.VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void RelativePaths_08()
+    {
+        // Unmapped file paths are not absolute. Relative path resolution is not performed.
```

Comment:
>Unmapped file paths are not absolute. Relative path resolution is not performed [](http://example.com/codeflow?start=11&length=79)

I don't understand this. It sounded like the goal of the change was to allow using relative paths. From the PR description:
> After this change we expect generators to start primarily using relative paths \[...]

Reply:
Path in the attribute is often relative, but we won't actually be able to resolve it when the containing syntax tree does not have an absolute path.